lib/bcoin.js

```diff
@@ -331,14 +400,24 @@ class LedgerBcoin {
    * @param {MTX} tx
    * @param {LedgerTXInput[]} ledgerInputs
    * @returns {TX}
+   * @throws {LedgerError}
+   * @throws {AssertionError}
    */
 
   async signTransaction(tx, ledgerInputs) {
     assert(MTX.isMTX(tx), 'Cannot use non-MTX tx for signing');
 
+    // Ledger should finish signing one transaction
+    // in order to sign another
+    const unlock = await this.txlock.lock();
```

Comment:
You can check this out on `bmutex` or in the Database code on `bcoin`.
Mutex/Lock provides a way to control concurrent to the library. If mutex is already locked another call should not be able to to execute code after this and will have to wait.

Unless I call `unlock` another part of the code that tries to call this will have to wait till current  lock `unlocks`. Limiting execution to only one party at a time